modules/library/main/src/main/java/org/geotools/data/FeatureSource.java

```diff
@@ -206,8 +206,36 @@
      *         or {@code -1} if this cannot be calculated.
      *
      * @throws IOException if there are errors getting the count
+     * @deprecated Please use {@link #count(Query)} instead
      */
-    int getCount(Query query) throws IOException;
+    @Deprecated
+    default int getCount(Query query) throws IOException {
```

Comment:
Document use of Integer.MAX_VALUE:
```
If this source contains more than Integer.MAX_VALUE elements, returns Integer.MAX_VALUE.
```

Reply:
I think it should throw an exception instead?

Reply:
If throwing it should really be [`java.lang.ArithmeticException`](https://docs.oracle.com/javase/8/docs/api/java/lang/ArithmeticException.html) to stay in line with what happens in eg. `java.lang.Math` when the result overflows an int, but method only declares `IOException` which seems counter-intuitive to throw. `ArithmeticException` being unchecked doesn't really need to be declared of course, but the unwitting programmer could be in for a surprise in that case.
That being said, I would prefer if it throws anything to having it return a magic value (even if it is `IOException` wrapping `ArithmeticException`)